### ISSUE TITLE
Refactor depth buffer rendering

### ISSUE THREAD

**Issue by mvaligursky** [playcanvas/engine#2953]
- refactor to better integrate with the engine
- implement support for WEBGL_depth_texture: https://github.com/playcanvas/engine/issues/2856
- fix: https://github.com/playcanvas/engine/issues/2932
- fix: ERROR: FRAMEBUFFER_INCOMPLETE_ATTACHMENT when resizing window and depth buffer is rendering. Repro: Post effects engine example with debug engine [this seems fixed in #4244]

**Comment by Maksims** [comment]
Not sure in details how it works now, but recently developer had to do this:

1. Render World Layer into RT.
2. Do few passes to blur texture into another RT.
3. Render Glass Layer (with previously blurred texture) from the respecting actual depth of the World Layer, so it occludes correctly.

And I think it was not smooth or difficult to do such thing efficiently, avoiding extra depth pre-pass on Glass Layer, so to avoid unnecessary drawcalls.

**Comment by YessicaSD** [comment]
Help!!! I have created SSAO post effect and It works fine in windows but in Mac it triggers the ERROR: FRAMEBUFFER_INCOMPLETE_ATTACHMENT (although it shows the effect)

**Comment by mvaligursky** [comment]
this is our attempt to have good quality SSAO .. https://github.com/playcanvas/engine/pull/3092

**Comment by YessicaSD** [comment]
I've made some changes and it doesn't crush but it seams that the depth buffer is not clearing.
I followed this project's structure -> https://playcanvas.com/project/642131/overview/depthmap-access

**Comment by mvaligursky** [comment]
if you need support, please post on the forum: http://forum.playcanvas.com/
This is to post issues with the engine code for us to work on.

**Comment by mvaligursky** [comment]
I tried to use Webgl1 extension `WEBGL_depth_texture` when refactoring the depth rendering in https://github.com/playcanvas/engine/pull/4244, to avoid having to re-render meshes from layers before the Depth layer.  
This can be done when the camera is rendering to a render target, as using the extension we can then copy the depth using shader into a texture (similarly as done in the Viewer: https://github.com/playcanvas/model-viewer/blob/main/src/read-depth.ts). The problem is when we're rendering straight into a default framebuffer - I don't see a way to copy out the depth from it, as it cannot be accessed as a texture and used in the shader. Specifically, I tried calling this to obtain the depth attachment (texture)
```
gl.bindFramebuffer(gl.FRAMEBUFFER, null);
const data = gl.getFramebufferAttachmentParameter(gl.FRAMEBUFFER, gl.DEPTH_ATTACHMENT,
                            gl.FRAMEBUFFER_ATTACHMENT_OBJECT_NAME);
```
but it results in
`WebGL: INVALID_OPERATION: getFramebufferAttachmentParameter: no framebuffer bound`
